vm.go

```diff
@@ -411,7 +499,10 @@ func (vm *vm) run() {
 		atomic.StoreUint32(&vm.interrupted, 0)
 		vm.interruptVal = nil
 		vm.interruptLock.Unlock()
-		panic(v)
+		panic(&uncatchableException{
```

Comment:
Should we panic with an `uncatchableException` on line 465 as well? Also, I'm not sure if we want this behavior or not, but as of now we always execute the `defer` above when we interrupt the VM with `interruptVal` set to a function. We've added that `defer` ourselves.

Reply:
Ah nice catch! Do you remember why we added the `defer` in the first place ?

Reply:
Ok I needed to take a deeper look here to get the full picture... The `defer` was added to recover from potential panics when executing the `interruptFunc`. From my understanding, we interrupt/pause the execution with a function to track the memory usage and to stop the execution when the context's Done channel is closed. 
When tracking memory usage, we don't actually want to stop the run. We just temporary pause the execution.

Reply:
Got it! I'll commit the latest changes coming from your comments then.